File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        # stack [73, 74, 75] decreasing towards the top
        # compare stack top vs curr
        # if curr is bigger than top, pop top and update
        stack = [0]
        ans = [0] * len(temperatures)
        for i in range(1, len(temperatures)):
            while stack and temperatures[i] > temperatures[stack[-1]]:
                ans[stack[-1]] = i - stack[-1]
                stack.pop()
            stack.append(i)
        return ans
        
        

File: 0739-daily-temperatures/0739-daily-temperatures.go
func dailyTemperatures(temperatures []int) []int {
    ans := make([]int, len(temperatures))
    stack := []int{}
    for i, v := range temperatures {
        for len(stack) != 0 && v > temperatures[stack[len(stack)-1]] {
            top := stack[len(stack)-1]
            stack = stack[:len(stack)-1]
            
            ans[top] = i - top
        }
        stack = append(stack, i)
    }
    return ans
}